lib/puppet/http/service/compiler.rb

```diff
@@ -13,8 +13,8 @@ def get_node(name, environment:, configured_environment: nil, transaction_uuid:
       with_base_url("/node/#{name}"),
       headers: headers,
       params: {
-        environment: environment,
-        configured_environment: configured_environment || environment,
+        environment: environment.name,
+        configured_environment: configured_environment || environment.name,
```

Comment:
Note: I need to change `configured_environment` to `configured_environment.name` as well.

Reply:
Ah, so [`configured_environment`](https://github.com/puppetlabs/puppet/blob/094781759ccb749bba2a50eb4293eeec1cf70ac6/lib/puppet/configurer.rb#L280) is a string, so I'd stick with that. Also I think we should pass an `environment` string, not `Puppet::Node::Environment` object into the various service methods. Perhaps raise if passed an enviromment parameter that isn't a string, so it's clear which type we support.